code/modules/paperwork/paper.dm

```diff
@@ -298,8 +311,40 @@
 
 		user << "<span class='notice'>You stamp the paper with your rubber stamp.</span>"
 
+	if(is_hot(P))
+		if(user.disabilities & CLUMSY && prob(10))
+			user.visible_message("<span class='warning'>[user] accidentally ignites themselves!</span>", \
+								"<span class='userdanger'>You miss the paper and accidentally light yourself on fire!</span>")
+			user.unEquip(P)
+			user.adjust_fire_stacks(1)
+			user.IgniteMob()
+			return
+
+		if(!(in_range(user, src))) //to prevent issues as a result of telepathically lighting a paper
+			return
+
+		user.unEquip(src)
+		user.visible_message("<span class='danger'>[user] lights [src] ablaze with [P]!</span>", "<span class='danger'>You light [src] on fire!</span>")
+		burn(showmsg = 0)
+
+
+
 	add_fingerprint(user)
 
+/obj/item/weapon/paper/fire_act()
+	burn(showmsg = 1)
+
+/obj/item/weapon/paper/proc/burn(var/burntime, var/showmsg = 1)
+	burntime = 70 //7 seconds, can be changed

```

Comment:
Your argument doesn't do anything.
Any value placed into the burntime argument becomes 70 now.